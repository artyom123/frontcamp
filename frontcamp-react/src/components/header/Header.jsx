import React from 'react';
import {
    makeStyles,
    Container,
    Grid,
} from '@material-ui/core';

import Search from './search/Search';
import Logo from './logo/Logo';

import stylesConstants from '../../styles/constants.module.scss';

const useStyles = makeStyles({
    grid: {
        backgroundColor: stylesConstants.secondaryDarkGray,
        paddingTop: '20px',
        paddingBottom: '20px',
    },
});

const Header = () => {
    const classes = useStyles();

    return (
        <Grid className={classes.grid}>
            <Container>
                <Logo />
                <Search/>
            </Container>
        </Grid>
    );
};

export default Header;
