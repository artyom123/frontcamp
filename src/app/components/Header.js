import Component from "./Component";
import List from "./List";
import { channels } from "../data/channels";

export default class Header extends Component {
    constructor() {
        super();
        this.header = document.querySelector(".header");
        this.list = new List();
    }

    render() {
        this.renderTitle();
        this.renderSelect();
    }

    renderTitle() {
        const h1 = this.createElement("h1");

        h1.textContent = "News headlines";
        this.header.appendChild(h1);
    }

    renderSelect() {
        const fragmentOptions = document.createDocumentFragment();
        const select = this.createElement("select", "select-channels");
        let optionFirst = null;

        channels.forEach((item, index) => {
            const option = this.createElement("option");
            const [text, valueItem] = item;

            option.textContent = text;
            option.value = valueItem;

            if (index === 0) {
                optionFirst = valueItem;
            }

            fragmentOptions.appendChild(option);
        });

        select.appendChild(fragmentOptions);
        this.header.appendChild(select);

        this.dispatchSelected(this.header, optionFirst);

        this.handleClick(select);
    }

    handleClick(element) {
        element.addEventListener("change", ({ target }) => {
            this.dispatchSelected(element, target.value);
        }, false);
    }

    dispatchSelected(element, detail) {
        element.dispatchEvent(new CustomEvent("selected_channel", { detail, bubbles: true }));
    }
}
