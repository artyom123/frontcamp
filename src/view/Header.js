import Component from "./component/Component";
import List from "./List";
import { channels } from "../common/data/channels";

const TEXT_HEAD_TITLE = "News headlines";

export default class Header extends Component {
    constructor() {
        super();
        this.header = document.querySelector(".header");
        this.list = new List();
    }

    render() {
        this.renderTitle();
        this.renderSelect();
    }

    renderTitle() {
        const h1 = this.createElement("h1");

        h1.textContent = TEXT_HEAD_TITLE;
        this.header.appendChild(h1);
    }

    renderSelect() {
        const fragmentOptions = document.createDocumentFragment();
        const select = this.createElement("select", "select-channels");
        let optionFirst = null;

        channels.forEach((item, index) => {
            const option = this.createElement("option");
            const [text, valueItem] = item;

            option.textContent = text;
            option.value = valueItem;

            if (index === 0) {
                optionFirst = valueItem;
            }

            fragmentOptions.appendChild(option);
        });

        select.appendChild(fragmentOptions);
        this.header.appendChild(select);

        this.dispatchSelected(this.header, optionFirst);

        this.handleClick(select);
    }

    handleClick(element) {
        element.addEventListener("change", ({ target }) => {
            this.dispatchSelected(element, target.value);
        }, false);
    }

    dispatchSelected(element, detail) {
        element.dispatchEvent(new CustomEvent("selected_channel", { detail, bubbles: true }));
    }
}
