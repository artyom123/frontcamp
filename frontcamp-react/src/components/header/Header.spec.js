import React from 'react'
import { shallow } from 'enzyme';
import Header from './Header';

describe('Header', () => {
    const header = shallow(<Header />)

    it('render initial', () => {
        expect(header).toMatchSnapshot();
    });
});