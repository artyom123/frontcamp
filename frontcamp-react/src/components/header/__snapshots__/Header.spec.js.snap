// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render initial 1`] = `ShallowWrapper {}`;
